test/object_utils_test.rb

```diff
@@ -13,9 +13,9 @@ class ObjectUtilsTest < TestCaseClass
     assert 1.unfriendly_id?
   end
 
-  test "numeric strings are neither friendly nor unfriendly" do
+  test "numeric strings are not friendly but are unfriendly" do
     assert_equal nil, "1".friendly_id?
-    assert_equal nil, "1".unfriendly_id?
+    assert_equal false, "1".unfriendly_id?
```

Comment:
Yeah. I didn't like changing this. The trouble is that the finder is only looking for `unfriendly_id? == true`. That is never true for any String so doing the `.to_i.to_s` check is just being ignored.
One option would be to change finder to call a different method like `definitely_not_friendly?` but I didn't want to add more things to Object. Possibly with refinements this wouldn't be an issue.